server/src/main/java/org/spine3/server/entity/Repository.java

```diff
@@ -294,4 +307,44 @@ public int getIndex() {
             return result;
         }
     }
+
+    /**
+     * An iterator of all entities from the storage.
+     */
+    private static class EntityIterator<I, E extends Entity<I, ?>>
```

Comment:
The idea is to use iterator over IDs instead of `loadAll()` methods in `RecordBasedRepository`.